docs/src/main/asciidoc/_admin-guide/subsystem-configuration/MicroProfile_Config_SmallRye.adoc

```diff
@@ -0,0 +1,106 @@
+[[MicroProfile_Config_SmallRye]]
+= MicroProfile Config Subsystem Configuration
+
+Support for https://microprofile.io/project/eclipse/microprofile-config[Eclipse MicroProfile Config] is provided by
+ the _microprofile-config-smallrye_ subsystem.
+
+[[required-extension]]
+== Required Extension
+
+This extension is included in the standard configurations included in the
+WildFly distribution.
+
+You can also add the extension to a configuration without it either by adding
+an `<extension module="org.wildfly.extension.microprofile.config-smallrye"/>`
+element to the xml or by using the following CLI operation:
+
+[source,ruby]
+----
+[standalone@localhost:9990 /]/extension=org.wildfly.extension.microprofile.config-smallrye:add
+----
+
+== Supported ConfigSources
+
+In addition to the default `ConfigSources` specified by the MicroProfile Config specification
+(environment variables, System properties and `META-INF/microprofile-config.properties` file), the
+`microprofile-config-smallrye` provides additional types of `ConfigSource`
+
+=== ConfigSource from Properties
+
+You can store properties directly in a `config-source` in the subsystem by using the `properties`
+attribute when you add the `config-source`:
+
+[source]
+----
+/subsystem=microprofile-config-smallrye/config-source=props:add(properties={"prop1" = "foo", "prop2" = "bar"})
+----
+
+This results in the XML configuration:
+
+[source,xml]
+----
+<subsystem xmlns="urn:wildfly:microprofile-config-smallrye:1.0">
+    <config-source name="props">
+        <property name="prop1" value="foo"/>
+        <property name="prop2" value="bar"/>
+    </config-source>
+</subsystem>
+----
+
+=== ConfigSource from Directory
```

Comment:
It seems that it is also possible to define ConfigSource by "class" name, right? Can you please describe it in this documentation?
```
/subsystem=microprofile-config-smallrye/config-source=aaaa:add(class={
```

Reply:
done